core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/listOfHashMaps/ListOfHashMapsUnitTest.java

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.list.listOfHashMaps;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+class ListOfHashMapsUnitTest {
+
+    @Test
+    void givenMaps_whenAddToList_thenListContainsMaps() {
+        List<HashMap<String, List<String>>> booksAuthorsMapsList = new ArrayList<>();
+        HashMap<String, List<String>> javaBooksAuthorsMap = new HashMap<>();
+        HashMap<String, List<String>> phpBooksAuthorsMap = new HashMap<>();
+
+        javaBooksAuthorsMap.put("Head First Java", Arrays.asList("Kathy Sierra", "Bert Bates"));
+        javaBooksAuthorsMap.put("Effective Java", Arrays.asList("Joshua Bloch"));
+        javaBooksAuthorsMap.put("OCA Java SE 8", Arrays.asList("Kathy Sierra", "Bert Bates", "Elisabeth Robson"));
+        phpBooksAuthorsMap.put("The Joy of PHP", Arrays.asList("Alan Forbes"));
+        phpBooksAuthorsMap.put("Head First PHP & MySQL", Arrays.asList("Lynn Beighley", "Michael Morrison"));
+
+        booksAuthorsMapsList.add(javaBooksAuthorsMap);
+        booksAuthorsMapsList.add(phpBooksAuthorsMap);
+
+        assertTrue(booksAuthorsMapsList.get(0)
+            .keySet()
+            .containsAll(javaBooksAuthorsMap.keySet()
+                .stream()
+                .collect(Collectors.toList())));
```

Comment:
You can simplify this code using ArrayList constructor so: 
```
        assertTrue(booksAuthorsMapsList.get(0)
            .keySet()
            .containsAll(new ArrayList<>(javaBooksAuthorsMap.keySet())));
```

Reply:
Done